### ISSUE TITLE
Which is the most promising architecture for image-to-sequence prediction?

### ISSUE THREAD

**Issue by einareinarsson** [keras-team/keras#8941]
I do experiments with a classic OCR task where the inputs are variable long handwritten digit sequence images of fix dimensions and the expected output is the character (digit) sequence. In the preprocessing phase I have encoded the training labels as padded label encoded fix vectors.

My first idea was to apply a cluster of CNN classifiers which produce a list output according to the maximal (padded) output vector length:

```
model_input=Input(shape=(img_channels, img_rows, img_cols))
x = Dropout(0.25)(model_input)
x = Convolution2D(16, (7, 7), padding ='valid', kernel_initializer='he_normal', activation="relu", data_format='channels_first')(model_input)
x = BatchNormalization()(x)
x = MaxPooling2D(pool_size=(2, 2), data_format='channels_first')(x)
x = Convolution2D(32, (7, 7), padding ='same', kernel_initializer='he_normal', activation="relu", data_format='channels_first')(x)
x = BatchNormalization()(x)
x = MaxPooling2D(pool_size=(2, 2), data_format='channels_first')(x)
x = Dropout(0.5)(x)
x = BatchNormalization()(x)
x = Flatten()(x)
x = Dense(256, activation="relu", kernel_constraint=maxnorm(3))(x)
x = Dropout(0.5)(x)
x = BatchNormalization()(x)
x = Dense(128, activation="relu", kernel_constraint=maxnorm(3))(x)
x = Dropout(0.5)(x)
conv_out = BatchNormalization()(x)

x1 = Dense(nb_classes, activation='softmax')(conv_out)
x2 = Dense(nb_classes, activation='softmax')(conv_out)
x3 = Dense(nb_classes, activation='softmax')(conv_out)
...
x13 = Dense(nb_classes, activation='softmax')(conv_out)

lst = [x1, x2, x3, ... x13]

model = Model(inputs=model_input, outputs=lst)
sgd = SGD(lr=lrate, momentum=0.9, decay=lrate/nb_epoch, nesterov=False)
model.compile(loss='categorical_crossentropy', optimizer=sgd, metrics=['accuracy'])
```

However, this approach seems to be inappropriate: the validation loss during training cannot go under 15.0 and indeed, at test evaluation the result is very disappointing (I have spent much time for fine tuning). My suspicion is that the main problem here is that the classifiers above are independent so albeit each of them could recognize certain digits on the picture the order of the sequence is messy (?) because the lack of sequential information.

Secondly I tried to apply [this CTC based tutorial](https://github.com/keras-team/keras/blob/master/examples/image_ocr.py) to my problem but it was even more disappointing (always predicted the padding zero as output) so at the moment I'm clueless where to continue. Has any of you any experience in similar problem?
